rwgame/GameConfig.cpp

```diff
@@ -88,50 +88,139 @@ struct BoolTranslator {
     }
 };
 
-bool GameConfig::readConfig(const std::string &path) {
-    pt::ptree config, defaultConfig;
+struct IntTranslator {
+    typedef std::string internal_type;
+    typedef int         external_type;
+    boost::optional<external_type> get_value(const internal_type &str) {
+        return boost::optional<external_type>(std::stoi(stripComments(str)));
+    }
+    boost::optional<internal_type> put_value(const external_type &i) {
+        return boost::optional<internal_type>(std::to_string(i));
+    }
+};
+
+std::string GameConfig::getDefaultINIString() {
+    std::string result;
+    parseConfig(ParseType::DEFAULT, "", ParseType::STRING, result);
+    return result;
+}
+
+bool GameConfig::parseConfig(
+    GameConfig::ParseType srcType, const std::string &source,
+    ParseType destType, std::string &destination)
+{
     bool success = true;
+    pt::ptree srcTree;
+
+    if ((srcType == ParseType::STRING) || (srcType == ParseType::FILE)) {
+        std::istream *istream = nullptr;
+        switch (srcType) {
+            case ParseType::STRING:
+                istream = new std::istringstream(source);
```

Comment:
For future OpenRW code it's preferable to avoid dynamic allocation, especially without using unique or shared ptr to avoid leaking memory if an exception gets thrown. This isn't a likely occurrence here but it's an approach openrw should adopt throughout.

Is it possible to do this without relying on a raw pointer? Perhaps refactoring the `if (istream != nullptr)` block into a function-local  lambda or a static member and passing a reference to an istream.